### PR TITLE
Add DocGen command

### PR DESCRIPTION
**Pull request by Taztingo**
## Description

Adds a hidden docgen command to generate provenance documentation.

closes: #1468

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote unit and integration [tests](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: cmd/provenanced/cmd/docgen.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+	"github.com/cosmos/cosmos-sdk/version"
+)
+
+var docGenCmdStart = fmt.Sprintf("%s docgen", version.AppName)
+
+const (
+	FlagMarkdown = "markdown"
+	FlagYaml     = "yaml"
+	FlagRest     = "rest"
```
Comment (iramiller) [inline cmd/provenanced/cmd/docgen.go:18]: The "rest" flag here was unexpected vs "rst" given how strongly  "rest" is associated with http services in my mind.

**Review by iramiller — DISMISSED** [review]
I like that this is a "hidden" command so it doesn't clutter things up.

File: cmd/provenanced/cmd/docgen.go (hunk 1 above)
Comment (Taztingo) [reply cmd/provenanced/cmd/docgen.go]: I agree. I updated it to "rst".

**Review by iramiller — APPROVED** [review]

**Review by nullpointer0x00 — APPROVED** [review]
Nice, this will be pretty useful!

**Pull request merged**